algorithms/kernel/pca/transform/oneapi/pca_transform_dense_default_batch_oneapi_impl.i

```diff
@@ -0,0 +1,270 @@
+
+#include "cl_kernels/pca_transform_cl_kernels.cl"
+
+using namespace daal::services;
+using namespace daal::oneapi::internal;
+using namespace daal::data_management;
+
+namespace daal
+{
+namespace algorithms
+{
+namespace pca
+{
+namespace transform
+{
+namespace oneapi
+{
+namespace internal
+{
+
+using namespace daal::oneapi::internal;
+
+template<typename algorithmFPType, transform::Method method>
+void TransformKernelOneAPI<algorithmFPType, method>::computeTransformedBlock
+        (uint32_t numRows, uint32_t numFeatures, uint32_t numComponents,
+         UniversalBuffer & dataBlock, UniversalBuffer & eigenvectors,
+         const services::Buffer<algorithmFPType> & resultBlock)
+{
+    BlasGpu<algorithmFPType>::xgemm(math::Layout::ColMajor, math::Transpose::Trans, math::Transpose::NoTrans, numComponents, numRows, numFeatures,
+        1.0, eigenvectors, numFeatures, 0, dataBlock, numFeatures, 0, 0.0, resultBlock, numComponents, 0);
+
+}
+
+
+template<typename algorithmFPType, transform::Method method>
+services::Status TransformKernelOneAPI<algorithmFPType, method>::computeInvSigmas(ExecutionContextIface& context,
+                                                                                  const KernelPtr & computeInvSigmasKernel,
+                                                                                  NumericTable* variances,
+                                                                                  const services::Buffer<algorithmFPType> & invSigmas,
+                                                                                  const size_t numFeatures)
+{
+    services::Status status;
+
+    BlockDescriptor<algorithmFPType> varBlock;
+    variances->getBlockOfRows(0, numFeatures, readOnly, varBlock);
+
+    KernelArguments args(2);
+    args.set(0, varBlock.getBuffer(), AccessModeIds::read);
+    args.set(1, invSigmas, AccessModeIds::write);
+    KernelRange range(numFeatures);
+    context.run(range, computeInvSigmasKernel, args, &status);
+    variances->releaseBlockOfRows(varBlock);
+    return status;
+}
+
+template<typename algorithmFPType, transform::Method method>
+services::Status TransformKernelOneAPI<algorithmFPType, method>::normalize(ExecutionContextIface& context,
+                                                                           const KernelPtr & normalizeKernel,
+                                                                           UniversalBuffer & copyBlock,
+                                                                           UniversalBuffer & rawMeans,
+                                                                           UniversalBuffer & invSigmas,
+                                                                           size_t numMeans,
+                                                                           size_t numInvSigmas,
+                                                                           const unsigned int maxWorkItemsPerGroup,
```

Comment:
Suppose not to pass kernel-specific params like `maxWorkItemsPerGroup`, `workItemsPerGroup`, `normalizeKernel` to the driver function, but define them inside it. If some of them are common for all drivers (like `maxWorkItemsPerGroup` probably), you can make it a private filed into the class